ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -2614,6 +2616,140 @@ prop_unit_4258 =
+               PeerSharingDisabled
+               [(1,1,Map.fromList [(RelayAccessAddress "0.0.0.0" 0,DoNotAdvertisePeer)])]
+               (Script (LedgerPools [] :| []))
+               PeerSelectionTargets {
+                 targetNumberOfRootPeers = 1,
+                 targetNumberOfKnownPeers = 1,
+                 targetNumberOfEstablishedPeers = 1,
+                 targetNumberOfActivePeers = 1,
+
+                 targetNumberOfKnownBigLedgerPeers = 0,
+                 targetNumberOfEstablishedBigLedgerPeers = 0,
+                 targetNumberOfActiveBigLedgerPeers = 0
+               }
+             (Script (DNSTimeout {getDNSTimeout = 10} :| [ ]))
+             (Script (DNSLookupDelay {getDNSLookupDelay = 0} :| []))
+             Nothing
+             False
+         , [ JoinNetwork 10
+           ])
+         ]
+
+      sim :: forall s . IOSim s Void
+      sim = diffusionSimulation (toBearerInfo (absNoAttenuation { abiInboundAttenuation  = SpeedAttenuation SlowSpeed (Time 20) 1000
+                                                                } ))
+                                diffScript
+                                iosimTracer
+
+      events :: [Events DiffusionTestTrace]
+      events = fmap ( Signal.eventsFromList
+                    . fmap (\(WithName _ (WithTime t b)) -> (t, b))
+                    )
+             . Trace.toList
+             . splitWithNameTrace
+             . Trace.fromList ()
+             . fmap snd
+             . Trace.toList
+             . fmap (\(WithTime t (WithName name b)) -> (t, WithName name (WithTime t b)))
+             . withTimeNameTraceEvents
+                @DiffusionTestTrace
+                @NtNAddr
+             . traceFromList
+             . fmap (\(t, tid, tl, te) -> SimEvent t tid tl te)
+             . take 125000
+             . traceEvents
+             $ runSimTrace sim
+
+   in conjoin
+    $ verify_consistency
+   <$> events
+
+  where
+    verify_consistency :: Events DiffusionTestTrace -> Property
+    verify_consistency events =
+      let govEstablishedPeersSig :: Signal (Set NtNAddr)
+          govEstablishedPeersSig =
+            selectDiffusionPeerSelectionState'
+              (EstablishedPeers.toSet . Governor.establishedPeers)
+              events
+
+          govConnectionManagerTransitionsSig :: [E (AbstractTransitionTrace NtNAddr)]
+          govConnectionManagerTransitionsSig =
+            Signal.eventsToListWithId
+            $ Signal.selectEvents
+                (\case
+                   DiffusionConnectionManagerTransitionTrace tr -> Just tr
+                   _                                            -> Nothing
+                ) events
+
+       in conjoin
+        $ map (\(E ts a) -> case a of
+                TransitionTrace addr (Transition _ TerminatedSt) ->
+                  eventually ts (Set.notMember addr) govEstablishedPeersSig
+                _ -> True -- TODO: Do the opposite
```

Comment:
What do you mean?

Reply:
That we should/could check that whenever there's a `Unnegotiated -> Negotiated Outbound` transition the peer is eventually added to the established set

Reply:
Please create an issue for it (an mention it in the todo).

Reply:
Then you can add it back to the merge queue.